.azure/prepare-docker.yml

```diff
@@ -0,0 +1,53 @@
+parameters:
+  image_name: ''
+  cache_key_version: ''
+  cache_key_folder: ''
+  use_cacheable: 'true'
+
+steps:
+  - ${{ if ne(parameters.image_name, '') }}:
+    - bash: mkdir -p $(Pipeline.Workspace)/cache/ci-${{ parameters.image_name }}/ccache
+      displayName: Prepare Docker Cache
+    
+    - task: CacheBeta@1
+      inputs:
+        key: '"docker-cache-ci-${{ parameters.image_name }}-v${{ parameters.cache_key_version }}" | scripts/${{ parameters.cache_key_folder }}/*.sh | Dockerfile'
+        path: $(Pipeline.Workspace)/cache/ci-${{ parameters.image_name }}
+        cacheHitVar: DOCKER_CACHE_RESTORED
+        restoreKeys: |
+          "docker-cache-ci-${{ parameters.image_name }}-v${{ parameters.cache_key_version }}"
+      displayName: Cache Docker history
+
+  - task: DockerInstaller@0
+    displayName: Docker Installer
+    inputs:
+      dockerVersion: 19.03.2
+      releaseType: stable
+
+  - bash:  |
+      set -ex
+      docker --version
+      mkdir -p ~/.docker/cli-plugins
+      curl --location https://github.com/docker/buildx/releases/download/v0.3.1/buildx-v0.3.1.linux-amd64 -o ~/.docker/cli-plugins/docker-buildx
+      chmod a+x ~/.docker/cli-plugins/docker-buildx
+    displayName: Install buildx
+
+  - ${{ if eq(parameters.use_cacheable, 'true') }}:
+    - bash:  |
+        set -ex
+        docker buildx create --name cacheable --use
+        docker buildx inspect --bootstrap
+        if [ -f $(Pipeline.Workspace)/cache/ci-${{ parameters.image_name }}/backup.tar.gz ]; then
+          docker buildx stop
+          docker run --rm \
+            --volumes-from buildx_buildkit_cacheable0 \
+            -v $(Pipeline.Workspace)/cache/ci-${{ parameters.image_name }}:/backup \
+            alpine /bin/sh -c "cd / && tar xf /backup/backup.tar.gz"
+        fi
+      displayName: Configure buildx cacheable context
+
+  - bash: echo "$DOCKER_PASSWORD" | docker login -u $(dockerhub.username) --password-stdin
```

Comment:
You are probably doing sufficiently custom Docker stuff that this wouldn't help you here, but you might want to look at https://github.com/jfpanisset/windows-container-test#using-a-service-connection-with-the-predefined-docker-task for an example of creating a service connection to Docker Hub which allows the use of the Docker@2 "login" command.

Reply:
Indeed, I did use this before but I found that without being admin on the ASWF Azure org it's a bit trickier to manage, and secret variables seem to work fine and securely as well so I went back to them. That said if we can find a good automated way of managing azure orgs on forked github repos then we could go back to service connections?